integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JMSInboundBrokerShutdownTest.java

```diff
@@ -0,0 +1,129 @@
+    )
+    public void testJMSInboundEndpointBehaviourWithBrokerShutdown() throws Exception {
+        addInboundEndpoint(AXIOMUtil
+                .stringToOM("<inboundEndpoint xmlns=\"http://ws.apache.org/ns/synapse\"\n"
+                        + "                 name=\"JMSInboundBrokerShutdown\"\n"
+                        + "                 sequence=\"requestHandlerSeq\"\n"
+                        + "                 onError=\"inFault\"\n"
+                        + "                 protocol=\"jms\"\n"
+                        + "                 suspend=\"false\">\n"
+                        + "    <parameters>\n"
+                        + "        <parameter name=\"interval\">10000</parameter>\n"
+                        + "        <parameter name=\"transport.jms.Destination\">brokershutdownqueue</parameter>\n"
+                        + "        <parameter name=\"transport.jms.CacheLevel\">0</parameter>\n"
+                        + "        <parameter name=\"transport.jms"
+                        + ".ConnectionFactoryJNDIName\">QueueConnectionFactory</parameter>\n"
+                        + "        <parameter name=\"java.naming.factory.initial\">org.apache.activemq.jndi"
+                        + ".ActiveMQInitialContextFactory</parameter>\n"
+                        + "        <parameter name=\"java.naming.provider.url\">tcp://localhost:61616</parameter>\n"
+                        + "        <parameter name=\"transport.jms.SessionAcknowledgement\">AUTO_ACKNOWLEDGE"
+                        + "</parameter>\n"
+                        + "        <parameter name=\"transport.jms.SessionTransacted\">false</parameter>\n"
+                        + "        <parameter name=\"transport.jms.ConnectionFactoryType\">queue</parameter>\n"
+                        + "    </parameters>\n"
+                        + "</inboundEndpoint>"));
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        String expectedErrorMessage = "Could not connect to broker URL: tcp://localhost:61616. Reason: java.net."
+                + "ConnectException: Connection refused (Connection refused)";
+
+        //Test behaviour when broker shuts down while the server is up
+        logViewerClient.clearLogs();
+        boolean assertAffected = false;
+        activeMQServer.stopJMSBroker();
+        long startTime = System.currentTimeMillis();
+        LogEvent[] remoteSystemLogs;
+        while (!assertAffected && (System.currentTimeMillis() - startTime) < 20000) {
+            remoteSystemLogs  = logViewerClient.getAllRemoteSystemLogs();
+            if (remoteSystemLogs != null) {
+                for (LogEvent logEvent : remoteSystemLogs) {
+                    if (logEvent.getMessage().contains(expectedErrorMessage)) {
+                        assertAffected = true;
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
See if you can use `Utils.assertIfSystemLogContains` instead of this block.

https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java#L135